pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -44,10 +44,33 @@ type preFilterState struct {
 	// criticalPaths[1].MatchNum is always greater or equal to criticalPaths[0].MatchNum, but
 	// it's not guaranteed to be the 2nd minimum match number.
 	TpKeyToCriticalPaths map[string]*criticalPaths
+	// TpKeyToDomainsNum is keyed with topologyKey, and valued with the number of domains.
+	TpKeyToDomainsNum map[string]int
 	// TpPairToMatchNum is keyed with topologyPair, and valued with the number of matching pods.
 	TpPairToMatchNum map[topologyPair]int
 }
 
+// minMatchNum returns the global minimum for the calculation of skew while taking MinDomains into account.
+func (s *preFilterState) minMatchNum(tpKey string, minDomains int32) (int, error) {
```

Comment:
I think I missed the validation of feature gate here. If feature is disabled, we should quick return.

Reply:
> we should quick return.

For performance? 
If the future gate is not enabled, do you think we shouldn’t use all the implementation for that future gate? (In other word, do we need to be that concerned about such detailed performance?) 
Actually, even if the feature is disabled, we set 1 to internal `minDomains`, and by definition of MinDomains, it shouldn't have effect on scheduling result when `minDomains=1`. So, this implement has no problem at least in behavior.

My current opinion is: 
using `pl.enableMinDomainsInPodTopologySpread` only in `filterTopologySpreadConstraints` func, set 1 to internal `minDomains` there, and in Filter, behaves without caring what value MinDomains is.

Reply:
You're right. Thanks for explaining.

Reply:
We should quick return when minDomains=1 because there could always be unforeseen bugs. If the feature is disabled, or they use the no-op value, we shouldn't run this code.